spec/setup/helpers/perf_helper_spec.rb

```diff
@@ -49,5 +72,429 @@ def options
+
+          it 'includes the dashboard and installs pe' do
+
+            expect(subject).to receive(:is_pre_aio_version?).and_return(test_is_pre_aio_version)
+            expect(subject).to receive(:use_meep?).with(master['pe_ver'] || options['pe_ver']).and_return(test_use_meep)
+
+            # TODO: is allow acceptable here? expect error with '5 times'
+            allow(subject).to receive(:master).and_return(master)
+
+            expect(subject).to receive(:compile_masters).and_return(compile_masters)
+            expect(subject).to receive(:dashboard).and_return(dashboard)
+
+            # TODO: handle custom answers
+
+            # TODO: handle pe_version master / options
+
+            # TODO: version_is_less??
+            expect(subject).to receive(:version_is_less).with(test_pe_ver, test_boundary_version).and_return(test_version_is_less)
+
+            # TODO: pre_config_hiera - with?
+            expect(subject).to receive(:pre_config_hiera)
+
+            # TODO: master['hieradata_dir_used_in_install']
+
+            expect(subject).to receive(:install_lei).once
+            subject.perf_install_pe
+
+          end
+
+        end
+
+        context 'when version is not less than 2016.2' do
+
+          test_pe_ver = '2016.3'
+          test_version_is_less = false
+
+          let!(:master) { {'pe_ver' => test_pe_ver} }
+          let!(:compile_masters) { {'pe_ver' => test_pe_ver} }
+          let!(:dashboard) { {'pe_ver' => test_pe_ver} }
+
+          it 'includes the dashboard and installs pe' do
+
+            expect(subject).to receive(:is_pre_aio_version?).and_return(test_is_pre_aio_version)
+            expect(subject).to receive(:use_meep?).with(master['pe_ver'] || options['pe_ver']).and_return(test_use_meep)
+
+            # TODO: is allow acceptable here? expect caused error with '3 times'
+            allow(subject).to receive(:master).and_return(master)
+
+            expect(subject).to receive(:compile_masters).and_return(compile_masters)
+            expect(subject).to receive(:dashboard).and_return(dashboard)
+
+            # TODO: handle custom answers?
+
+            # TODO: handle pe_version master / options?
+
+            # TODO: version_is_less??
+            expect(subject).to receive(:version_is_less).with(test_pe_ver, test_boundary_version).and_return(test_version_is_less)
+
+            expect(subject).to receive(:install_lei).once
+            subject.perf_install_pe
+
+          end
+
+        end
+
+      end
+
+    end
+
+    context 'when pe is a pre-aio version' do
+
+      test_is_pre_aio_version = true
+
+      it 'installs pe without including the dashboard' do
+        expect(subject).to receive(:is_pre_aio_version?).and_return(test_is_pre_aio_version)
+        expect(subject).to receive(:install_lei)
+        subject.perf_install_pe
+      end
+
+    end
+
+  end
+
+  describe '#has_cent7_repo?' do
+
+    # TODO: stub Net::HTTP to provide the expected response or test the actual URLs?
```

Comment:
I would definitely stub this, but leaving a TODO is fine for now.

Reply:
It is actually stubbing it here and elsewhere; I just wanted to make sure that was the right thing to do.

Reply:
Perfect